src/asset-management/EulerV2Manager.sol

```diff
@@ -128,62 +108,46 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
                                 HELPER FUNCTIONS
     //////////////////////////////////////////////////////////////////////////*/
 
-    function _increaseShares(IAssetManagedPair aPair, IERC20 aToken, IERC20 aAaveToken, uint256 aAmount)
+    function _increaseShares(IAssetManagedPair aPair, IERC20 aToken, IERC4626 aVault, uint256 aAmount)
         private
         returns (uint256 rShares)
     {
-        uint256 lTotalShares = totalShares[aAaveToken];
-        if (totalShares[aAaveToken] == 0) {
-            rShares = aAmount;
-        } else {
-            rShares = aAmount * lTotalShares / aAaveToken.balanceOf(address(this));
-        }
+        rShares = aVault.previewDeposit(aAmount);
```

Comment:
What is previewDeposit, does it actually deposit?

Reply:
this function simulates a deposit, and tells us how many shares it returns but doesn't actually deposit the assets. Honestly we can just skip this step, and do the deposit right away, and keep track of the shares from the return value of `deposit()`

Reply:
Sounds good